up42/asset.py

```diff
@@ -25,35 +25,30 @@ class Asset:
     ```
     """
 
-    def __init__(self, auth: Auth, asset_id: str, asset_info: Optional[dict] = None):
+    def __init__(
+        self,
+        auth: Auth,
+        asset_id: Optional[str] = None,
+        asset_info: Optional[dict] = None,
+    ):
+        if asset_id is not None and asset_info is not None:
+            raise ValueError("asset_id and asset_info cannot be provided simultaneously.")
+        if asset_id is None and asset_info is None:
+            raise ValueError("Either asset_id or asset_info should be provided in the constructor.")
         self.auth = auth
-        self.asset_id = asset_id
+        self.info = asset_info or self._get_info(asset_id)
```

Comment:
This is not covered

Reply:
I think this is covered by adding the `test_should_initialize_with_provided_info`, and `test_should_initialize_with_retrieved_info`